### ISSUE TITLE
Add setting to poll layers for new data (CSV)

### ISSUE THREAD

**Issue by axman6** [TerriaJS/terriajs#1494]
On AREMI, we have a service which produces a CSV which is updated every 5 minutes. We would like to have Terria fetch this every N minutes from the same URL and update the information shown on the map when it's enabled so we can have live feature info and map rendering. @RacingTadpole has been working on related periodic data updates for the Charts work (#1278).

**Comment by RacingTadpole** [comment]
Breaking this down we want to:
- Poll for new data in a CsvCatalogItem (either replace or update the existing data), and:
  - a) Redisplay it on the map (if it is spatial).
  - b) Update any displayed features in the Feature Info Panel (if it is spatial).
  - c) Update the Chart Panel (if it is not spatial).
- Poll for new csv data in any `<chart>` tags in the Feature Info Panel, and update those charts.

The interface for the first would be something like `pollUrl` and `pollSeconds` (and maybe a boolean `pollReplace`?) in the CsvCatalogItem json. I have this, and parts (a) and (c), working in the `newui_poll` branch (with replace=true).

The equivalent interface for the second would be to write eg. `<chart poll-sources="..., ..." poll-seconds="60">`. The poll-sources are just like the `sources` attribute (ie. comma-separated URLs for data at each available time range). When the chart is expanded, it transfers the poll-seconds and the relevant poll-source onto the newly created (non-spatial) CsvCatalogItem.

**Comment by RacingTadpole** [comment]
Also, (b) is something new: we'll need to link rows of the old and the updated data tables.

#### For lat/lon files

Suppose the old data was:

```
lat,lon,name,value
30,130,Foo,5
40,140,Bar,6
```

and the new data is:

```
lat,lon,name,value
30,130,Foo,8
45,140,Bar,4
40,140,La,1
```

If the user had the Feature Info Panel open on "Bar" at (40,140), should it be updated to now show "Bar" in its new location, or "La" at the original location (40,140)? I was going to say it should maintain the row with the same lat/lon, but then it wouldn't track a car's real-time location (or any mobile object). And immobile things shouldn't be changing name anyway.

A proposal:
- Instead of guessing, require the user to specify one or more id columns via `pollIdColumns` (eg. so you could specify ["lat","lon"] or ["name"] or something more complex).

If the data has a `time` column as well, things get more complicated. Time-dependent data is more likely to be "updated" than "replaced", I think - in which case it should "just work"...

#### For region-mapped files:

This is easier: match rows across the two versions of the data by region. So if your Feature Info Panel is open to SA3 code 10102, just keep it there.

**Comment by RacingTadpole** [comment]
Allowing the polled data to contain "updates" to the original data (rather than replacing it entirely) brings up the same issues as (b) above, at least for spatial data.

Updating is more straightforward for non-spatial data, assuming we know the "x" column - if there is a matching x-value, replace its y-value, otherwise add the new (x, y).

To keep it simple for now, I propose we only allow replacement, not updates. We can always extend it later.

**Comment by axman6** [comment]
Some of these more complex uses are probably better suited to CZML - as a first pass I'd be ok with replacing the visualised data wholesale for our use case of the AEMO data.

**Comment by stevage** [comment]
Arriving a bit late here, but "reload layer every X minutes" sounds potentially useful for almost every catalog item type, right - not just CSV? Can easily imagine using something like that for .geojson, for instance.

**Comment by RacingTadpole** [comment]
Here's an interesting decision.

Say we have a csv file which is polled and updated every 5 minutes, ie. the points on the map are updated with the new data, and any open feature info is also updated.

Now, that feature info contains a `<chart src="xyz.csv">`, which gets its data from a URL. For AEMO, that URL (eg. `xyz.csv`) doesn't ever change - but it is an API call which gets the latest data, so the data at that URL _does_ change.

The question is, should `<chart src="xyz.csv">`:
i) Re-poll `xyz.csv` at the same rate as its parent csv file, in case its data has changed?
ii) Or follow the principle that none of its attributes have changed, so there is no need to update?

Note if the 5-minute update had changed the URL to `<chart src="abc.csv">` then I think it should automatically reload the chart data from the new URL.

Also note that ideally we could also update the `<chart>` in the feature info, without needing to update the parent data, eg. via `<chart poll-src="xyz.csv" poll-seconds="300">`. So we could implement (ii) by requiring an explicit `poll-seconds` attribute in `<chart>` in the AEMO feature info template.

I think (ii) has one advantage over (i):
- If the data at the URL (eg. `xyz.csv`) is static, then there's no unnecessary reloading of the data.

**Comment by kring** [comment]
Fixed by #1756.
